tests/draft4/format.json

```diff
@@ -1,217 +1,247 @@
 [
     {
-        "description": "validation of e-mail addresses",
-        "schema": {"format": "email"},
+        "description": "email format",
+        "schema": { "format": "email" },
         "tests": [
             {
-                "description": "ignores integers",
+                "description": "all string formats ignore integers",
                 "data": 12,
                 "valid": true
             },
             {
-                "description": "ignores floats",
+                "description": "all string formats ignore floats",
                 "data": 13.7,
                 "valid": true
             },
             {
-                "description": "ignores objects",
+                "description": "all string formats ignore objects",
                 "data": {},
                 "valid": true
             },
             {
-                "description": "ignores arrays",
+                "description": "all string formats ignore arrays",
                 "data": [],
                 "valid": true
             },
             {
-                "description": "ignores booleans",
+                "description": "all string formats ignore booleans",
                 "data": false,
                 "valid": true
             },
             {
-                "description": "ignores null",
+                "description": "all string formats ignore nulls",
                 "data": null,
                 "valid": true
+            },
+            {
+                "description": "invalid email string is only an annotation by default",
```

Comment:
I'm not sure we should add this here, but I need to double check that I'm not misremembering --

Having format be an annotation by default is required only as of draft 7 I think -- in previous drafts, implementations are free to make it an assertion by default if I'm not mistaken.

Reply:
yes that's fair. the spec in draft7 says that implementations MAY choose to assert, but it is ambiguous about what the default behaviour should be, so I removed those bits from drafts4,6,7.

Reply:
ah damn, I keep forgetting about draft3 (re https://github.com/json-schema-org/JSON-Schema-Test-Suite/commit/6327a3bcd2b2991e9be209f71230f6b57f0f9460) :D. I wonder if we should just delete all those tests, since we don't even have the spec posted for draft3 anymore so it's pretty hard to check whether tests are actually valid there.